internal/cel/compiler.go

```diff
@@ -129,6 +130,65 @@ func createMutationFunction(name string, mutationType MutationType, returnType *
 	)
 }
 
+// createResourceMutationFunction creates a CEL function for resource mutations that accepts string key and int value
+func createResourceMutationFunction(name string, mutationType MutationType, returnType *cel.Type) cel.EnvOption {
+	return cel.Function(
+		name,
+		cel.Overload(
+			name+"_string_int_to_mutation",
+			[]*cel.Type{cel.StringType, cel.IntType},
+			returnType,
+			cel.BinaryBinding(func(lhs, rhs ref.Val) ref.Val {
+				key, keyOk := lhs.Value().(string)
+
+				if !keyOk {
+					return types.NewErr("%s function requires string key argument", name)
+				}
+
+				if key == "" {
+					return types.NewErr("%s key cannot be empty", name)
+				}
+
+				intValue, intValueOk := rhs.Value().(int64)
+
+				if !intValueOk {
+					return types.NewErr("%s function requires int value argument", name)
+				}
+
+				// Validate that the value is positive (non-negative)
+				if intValue < 0 {
+					return types.NewErr("%s value must be positive (>= 0), got %d", name, intValue)
+				}
+
+				// Convert int value to string for storage
+				value := fmt.Sprintf("%d", intValue)
+
+				// Validate key using annotation validation since resource mutations create annotations
+				err := validateKey(key, "resource annotation")
+				if err != nil {
+					return types.NewErr("%s key validation failed: %v", name, err)
+				}
+
```

Comment:
minor: no need to convert value to string if key is invalid

```suggestion
				// Validate key using annotation validation since resource mutations create annotations
				err := validateKey(key, "resource annotation")
				if err != nil {
					return types.NewErr("%s key validation failed: %v", name, err)
				}

				// Convert int value to string for storage
				value := fmt.Sprintf("%d", intValue)

```